projects/optic/src/commands/diff/upload-diff.ts

```diff
@@ -30,34 +29,30 @@ export async function uploadDiff(
     ? ora({ text: `Uploading diff...`, color: 'blue' })
     : null;
 
-  const opticUrl: string | null =
-    specs.to.jsonLike[OPTIC_URL_KEY] ??
-    specs.from.jsonLike[OPTIC_URL_KEY] ??
-    null;
-  const specDetails = opticUrl ? getApiFromOpticUrl(opticUrl) : null;
   // We upload a spec if it is unchanged in git and there is an API id on the spec
   let baseSpecId: string | null = null;
   let headSpecId: string | null = null;
   if (specs.from.context && specDetails) {
-    const tags =
-      specs.from.context.vcs === VCS.Git
-        ? [`git:${specs.from.context.sha}`]
-        : [];
-    baseSpecId = await uploadSpec(specDetails.apiId, {
-      spec: specs.from,
-      client: config.client,
-      tags,
-      orgId: specDetails.orgId,
-    });
+    if (specs.from.context.vcs === VCS.Git) {
+      const tags = [`git:${specs.from.context.sha}`];
+      baseSpecId = await uploadSpec(specDetails.apiId, {
+        spec: specs.from,
+        client: config.client,
+        tags,
+        orgId: specDetails.orgId,
+      });
+    } else if (specs.from.context.vcs === VCS.Cloud) {
```

Comment:
Fixes the upload flow when using `--compare-from cloud:tag` - since we'll set the spec id to whatever we downloaded from cloud